tool/clean/clean_host/clean_host.go

```diff
@@ -49,6 +49,7 @@ func terminateInstances(cxt context.Context, ec2client *ec2.Client) {
 		"IntegrationTestBase",
 		"CWADockerImageBuilderX86",
 		"CWADockerImageBuilderARM64",
+		"cwagent-integ-test-ec2*",
```

Comment:
When does this cleaner run?
Is it possible the cleaner runs while tests are running and therefore cause false test failures?

Reply:
There's a check in the cleaner code that will skip instances that are newer than a day, so shouldn't affect running tests.

Reply:
This cleaner run every day at 0:00 UTC and it's impossible for tests to be running and get deleted incorrectly because the cleaner will check if the instance has been running for more than a than day, and if so it will terminate it. I don't think we have any test besides canary and soaking tests that take that long but those should be separated from t his process.

The issue with cleanup after test calls is sometimes a failure or cancellation will completely skip the terraform destroy even if a separate job was created for it. A good example is [this run](https://github.com/zhihonl/amazon-cloudwatch-agent/actions/runs/5017163339/jobs/8995177970). The workflow was canceled but the instance still remained on my personal account even though we have a separate destroy job after `terraform apply` as an attempt to mitigate it.